docs/spec.openapi.yaml

```diff
@@ -0,0 +1,627 @@
+    post:
+      tags:
+        - Files
+      description: Share a file with a user
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/shareWithReq'  
+      responses:
+        '200':
+          description: File shared successfully
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/statusResponse'
+        '400':
+          description: Bad request
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/statusResponse'
+        '401':
+          description: Not authorized
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/statusResponse'
+        '404':
+          description: File not found
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/statusResponse'
+        '500':
+          description: Internal error
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/statusResponse'
+                
+  /files/unshare:
+    post:
+      tags:
+        - Files
+      description: Unshare a file with a user
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/unshareWithReq'  
+      responses:
+        '200':
+          description: File unshared successfully
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/statusResponse'
+        '400':
+          description: Bad request
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/statusResponse'
+        '401':
+          description: Not authorized
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/statusResponse'
+        '404':
+          description: File not found
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/statusResponse'
+        '500':
+          description: Internal error
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/statusResponse'
+              
+  /files/sharedwithwho:
+    get:
+      tags:
+        - Files
+      description: Get the list of users the file is shared with
+      parameters:
+        - in: query
+          name: fileUUID
+          required: true
+          schema:
+            type: string  
+      responses:
+        '200':
+          description: List of users the file is shared with
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/sharedWithWhoResponse'  
```

Comment:
@Woynert, should this return the usernames instead of the UUIDs?

Reply:
You're right usernames are the primary way we identify users, I'm going to update the SOAP interfaces.